### PR TITLE
Fix saved credentials when using multiple users

### LINKED ISSUE #1636: Roku Jellyfin 2.0.1 unable to remember different profiles
## Software Versions

- Jellyfin Server Version:  10.8.13
- Roku Client Version:  2.0.1

## Describe the bug

After updating from 2.0.0 Jellyfin Roku client to 2.0.1, the client does not save the passwords of different user profiles on the same server. 

## How To Reproduce

1. Create two user accounts "A" and "B"
2. Login to "A" on Jellyfin Roku. ("Save Credentials" is selected)
3. Press "*" on the remote, "Change User"
4. Login to "B" ("Save Credentials" is selected)
5. Press "*" on the remote, "Change User", and reselect "A"
6. The 2.0.1 Roku Client now requires to type in the password for "A" again. This does not happen in 2.0.0 (The client will automatically login to "A" again.)

## Logs
The following are the Logs from the SERVER

[2024-01-12 15:30:58.084 +00:00] [INF] [3] Jellyfin.Server.Implementations.Users.UserManager: Authentication request for "User A" has succeeded.
[2024-01-12 15:30:58.084 +00:00] [INF] [3] Emby.Server.Implementations.Session.SessionManager: Current/Max sessions for user "User A": 2/0
[2024-01-12 15:30:58.086 +00:00] [INF] [3] Emby.Server.Implementations.Session.SessionManager: Creating new access token for user e3c01808-2602-48c5-875a-b1490b5f6e8d
[2024-01-12 15:31:26.885 +00:00] [INF] [51] Jellyfin.Api.Auth.CustomAuthenticationHandler: "CustomAuthentication" was not authenticated. Failure message: "Invalid token."
[2024-01-12 15:31:26.885 +00:00] [INF] [51] Jellyfin.Api.Auth.CustomAuthenticationHandler: "CustomAuthentication" was not authenticated. Failure message: "Invalid token."
[2024-01-12 15:31:26.887 +00:00] [INF] [51] Jellyfin.Api.Auth.CustomAuthenticationHandler: AuthenticationScheme: "CustomAuthentication" was challenged.
[2024-01-12 15:31:26.924 +00:00] [INF] [29] Jellyfin.Api.Auth.CustomAuthenticationHandler: "CustomAuthentication" was not authenticated. Failure message: "Invalid token."
[2024-01-12 15:31:27.064 +00:00] [ERR] [29] Jellyfin.Server.Implementations.Users.UserManager: Error authenticating with provider "Default"
MediaBrowser.Controller.Authentication.AuthenticationException: Invalid username or password
   at Jellyfin.Server.Implementations.Users.DefaultAuthenticationProvider.Authenticate(String username, String password, User resolvedUser)
   at Jellyfin.Server.Implementations.Users.UserManager.AuthenticateWithProvider(IAuthenticationProvider provider, String username, String password, User resolvedUser)
[2024-01-12 15:31:27.077 +00:00] [INF] [29] Jellyfin.Server.Implementations.Users.UserManager: Authentication request for "User B" has been denied (IP: "192.168.0.7").
[2024-01-12 15:31:27.079 +00:00] [ERR] [29] Jellyfin.Server.Middleware.ExceptionMiddleware: Error processing request: "Invalid username or password entered". URL "POST" "/Users/AuthenticateByName".

## Connection Information

- Is server local or remote: Local
- Is server connection HTTP or HTTPS: HTTP

### PR DESCRIPTION
**Pull request by cewert**
Saved credentials currently only works right with one user. Once a second user auths using the device, the first user's saved auth token will stop working. This fixes that

NOTE: It only breaks for user's with a password. If your public user doesn't have a password, you won't notice this behavior because the client will automatically try to use a blank password to authenticate when you select your public user.

## Changes
- Update `SetServerDeviceName()` so that global gets updated as expected.
- Small QOL refactor to automatically update the auth header and session.user.friendlyName whenever we manually update `m.global.session.user.name` i.e. using `session.user.Update("name", userSelected)`  will automatically format and save the friendlyName and update the auth string

## Issues
Fixes #1636

### CODE REVIEW

File: source/utils/session.bs [hunk 1]
```diff
@@ -119,12 +119,25 @@ namespace session
...
             ' validate parameters
             if key = "" or value = invalid then return
+
             ' make copy of global user session
             tmpSessionUser = m.global.session.user
             ' update the temp user array
             tmpSessionUser[key] = value
+
+            ' keep friendlyName in sync
+            if key = "name"
```
Comment (1hitsong) [inline source/utils/session.bs]: String comparisons should force case

File: source/utils/session.bs [hunk 2]
```diff
@@ -119,12 +119,25 @@ namespace session
...
+                regex = CreateObject("roRegex", "[^a-zA-Z0-9\ \-\_]", "")
+                tmpSessionUser["friendlyName"] = regex.ReplaceAll(value, "")
+            end if
+
             ' update global user session using the temp array
             session.Update("user", tmpSessionUser)
+
+            ' keep auth header in sync
+            if key = "name"
```
Comment (1hitsong) [inline source/utils/session.bs]: String comparisons should force case

**Review by 1hitsong — CHANGES_REQUESTED** [review]
Bug recreated and confirmed this PR fixes it.

File: source/utils/session.bs (hunk 1 above)
Comment (cewert) [reply source/utils/session.bs]:
```suggestion
            if LCase(key) = "name"
```

File: source/utils/session.bs (hunk 2 above)
Comment (cewert) [reply source/utils/session.bs]:
```suggestion
            if LCase(key) = "name"
```

**Review by 1hitsong — APPROVED** [review]

**Pull request merged**